File: DesafiosPython/URI1047.py
hora = input().split(' ')
hi = int(hora[0])
mi = int(hora[1])
hf = int(hora[2])
mf = int(hora[3])
mti = int(hi*60+mi)
mtf = int(hf*60+mf)

if mti >= mtf:
    hf = hf+24
    mti = hi*60+mi
    mtf = hf*60+mf
    mtf = mtf - mti
    h = mtf // 60
    m = mtf % 60
    print(f'O JOGO DUROU {h} HORA(S) E {m} MINUTO(S)')
else:
    mtf = mtf-mti
    h = mtf // 60
    m = mtf % 60
    print(f'O JOGO DUROU {h} HORA(S) E {m} MINUTO(S)')

File: DesafiosJava/URI1047.java
import java.util.Scanner;

public class URI1047{
    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);
        String[] valores = entrada.nextLine().split(" ");
        entrada.close();
        int horaIni = Integer.parseInt(valores[0]);
        int minIni = Integer.parseInt(valores[1]);
        int horaFim = Integer.parseInt(valores[2]);
        int minFim = Integer.parseInt(valores[3]);

        int tempoEmMinutosInicio = horaIni * 60 + minIni;
        int tempoEmMinutosFim = horaFim * 60 + minFim;
        int duracaoJogo;

        if (tempoEmMinutosFim <= tempoEmMinutosInicio){
            duracaoJogo = (24 * 60 - tempoEmMinutosInicio) + tempoEmMinutosFim;
        }
        else {
            duracaoJogo = tempoEmMinutosFim - tempoEmMinutosInicio;
        }

        System.out.printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n ", duracaoJogo/60, duracaoJogo%60);
    }
}